src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -466,11 +467,278 @@ type Moon {
+                configProvider,
+                graphQLQueryName,
+                graphQLQuery);
+
+            // Assert
+            Assert.IsTrue(SqlTestHelper.JsonStringsDeepEqual($"{restBookContents}", initialRestContents));
+            SqlTestHelper.PerformTestEqualJsonStrings(BOOK_DBO_CONTENTS, initialGQLContents.GetProperty("items").ToString());
+
+            // Arrange
+            GenerateConfigFile(
+                connectionString: $"{GetConnectionStringFromEnvironmentConfig(TestCategory.MSSQL)}",
+                restPath: restPath,
+                gQLPath: gQLPath);
+            System.Threading.Thread.Sleep(1000);
+
+            // Act
+            result = await httpClient.GetAsync($"{restPath}/Book");
+            string reloadRestContent = await result.Content.ReadAsStringAsync();
+            JsonElement reloadGQLContents = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                httpClient,
+                configProvider,
+                graphQLQueryName,
+                graphQLQuery);
+
+            // Assert
+            Assert.IsTrue(SqlTestHelper.JsonStringsDeepEqual(restBookContents, reloadRestContent));
+            SqlTestHelper.PerformTestEqualJsonStrings(BOOK_DBO_CONTENTS, reloadGQLContents.GetProperty("items").ToString());
+        }
+
+        /// <summary>
+        /// Start DAB with pre-formed config file, assert that requests sent to
+        /// the rest and gql endpoints result in result status OK. Hot-reload
+        /// the config and set rest and graphql enabled to false, and assert
+        /// that requests sent to the rest and gql endpoints result in result
+        /// status NotFound.
+        /// </summary>
+        /// <returns></returns>
+        [TestCategory(MSSQL_ENVIRONMENT)]
+        [TestMethod("Hot-reload rest and gql enabled.")]
+        public async Task HotReloadConfigRuntimeEnabledEndToEndTest()
+        {
+            // Arrange
+            string restEnabled = "false";
+            string gQLEnabled = "false";
+            string query = HOT_RELOAD_GQL_QUERY;
+            object payload =
+                new { query };
+
+            GenerateConfigFile(connectionString: $"{GetConnectionStringFromEnvironmentConfig(TestCategory.MSSQL)}");
+            TestServer server = new(Program.CreateWebHostBuilder(new string[] { "--ConfigFileName", HOT_RELOAD_CONFIG_FILE_NAME  }));    
+            HttpClient httpClient = server.CreateClient();
+            HttpRequestMessage request = new(HttpMethod.Post, "/graphQL")
+            {
+                Content = JsonContent.Create(payload)
+            };
+
+            // Act
+            HttpResponseMessage restResult = await httpClient.GetAsync("/rest/Book");
+            HttpResponseMessage gQLResult = await httpClient.SendAsync(request);
+
+            // Assert
+            Assert.AreEqual(HttpStatusCode.OK, restResult.StatusCode);
+            Assert.AreEqual(HttpStatusCode.OK, gQLResult.StatusCode);
```

Comment:
nit: Are these really act and assert on the changes you want to validate? These are technically the "Arrange" section setting up the test so that you can validate the side effects of the hot reload.